pkg/sdk/helpers_proposal.go

```diff
@@ -0,0 +1,61 @@
+package sdk
+
+import (
+	"context"
+)
+
+// validatableOpts is just a proposal how we can remove some of the boilerplate.
+type validatableOpts interface {
+	// validateProp will be renamed if accepted.
+	// Having the name `validate` now would make all other `...Opts` structs validatableOpts because of Go nature.
+	validateProp() error
+}
+
+// validateAndExec is just a proposal how we can remove some of the boilerplate.
+func validateAndExec(client *Client, ctx context.Context, opts validatableOpts) error {
```

Comment:
Yeah, let's discuss it further, because I kinda like the idea. We have been also thinking about different possibilities here, I have sent you a doc, which summarised it.